hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/HttpClientsFactory.java

```diff
@@ -63,18 +64,13 @@ public HttpClient createClientForHttp2() {
         http2Client.setExecutor(executor);
         HttpClientTransportOverHTTP2 transport = new HttpClientTransportOverHTTP2(http2Client);
 
-        HttpClient client = new HttpClient(transport, createSslContextFactory());
+        HttpClient client = sslContextFactoryProvider.provideSslContextFactory()
+                .map(sslContextFactory -> new HttpClient(transport, sslContextFactory))
+                .orElseThrow(() -> new IllegalStateException("Cannot create http/2 client due to lack of ssl context factory"));
```

Comment:
It is reasonable behaviour because http/2 client should be created only when both ssl and http/2 are enabled.